### DISCUSSION TITLE
Does Prisma.sql support object mapping?

### DISCUSSION THREAD

**Discussion by mihok** [prisma/prisma#13207] [Q&A]
I'm in the process of adding Prisma to a large backend service, so I've been using $queryRaw as a stop gap until we can fully convert all of the code.

We have statements like:
```javascript
const query = `INSERT INTO blah SET = ?`
```

Which we then pass in a large object of values for blah, something along the lines of:
```javascript
const params = {
  param1: 'blah',
  param2: 42,
  ...,
  paramN: 'N',
}
```

This ends up getting destructured from the query above and the params to look something alonng the lines of:

```sql
INSERT INTO blah SET param1 = 'blah', param2 = 42, ..., paramN = 'N';
```

Is this possible to do with Prisma.sql and/or $queryRaw?

**Comment by nurul3101** [comment]
Hey @mihok 👋 

[queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) allows you to pass a template literal where you can easily insert your [variables](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#using-variables). In turn, Prisma creates prepared statements that are safe from SQL injections, like

```javascript
const email = 'test@prisma.io'
const result = await prisma.$queryRaw`SELECT * FROM User WHERE email = ${email}`
```
So you could declare all the variables and then pass them in your insert statement, but there isn't any functionality which would automatically destructure the object and add it in raw query.

**Reply by mihok** [reply]
The issue with that is that that means rewriting each query and most of the objects contain 10-20 parameters.. I was really hoping to not have to do that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
